src/resolve/test-resolved-packet.c

```diff
@@ -0,0 +1,45 @@
+/***
+  This file is part of systemd
+
+  Copyright 2017 Zbigniew Jędrzejewski-Szmek
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include "log.h"
+#include "resolved-dns-packet.h"
+
+static void test_dns_packet_new(void) {
+        size_t i;
+
+        for (i = 0; i < DNS_PACKET_SIZE_MAX + 2; i++) {
+                _cleanup_(dns_packet_unrefp) DnsPacket *p = NULL;
+
+                assert_se(dns_packet_new(&p, DNS_PROTOCOL_DNS, i) == 0);
+
+                log_debug("dns_packet_new: %zu → %zu", i, p->allocated);
+                assert_se(p->allocated >= MIN(DNS_PACKET_SIZE_MAX, i));
```

Comment:
There is a warning:
```sh
src/resolve/test-resolved-packet.c: In function ‘test_dns_packet_new’:
  CC       src/resolve/test_resolved_packet-resolved-dns-question.o
./src/basic/macro.h:187:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 UNIQ_T(A,aq) < UNIQ_T(B,bq) ? UNIQ_T(A,aq) : UNIQ_T(B,bq); \
                              ^
./src/basic/macro.h:44:44: note: in definition of macro ‘_unlikely_’
 #define _unlikely_(x) (__builtin_expect(!!(x),0))
                                            ^
./src/basic/macro.h:235:25: note: in expansion of macro ‘assert_message_se’
 #define assert_se(expr) assert_message_se(expr, #expr)
                         ^
src/resolve/test-resolved-packet.c:32:17: note: in expansion of macro ‘assert_se’
                 assert_se(p->allocated >= MIN(DNS_PACKET_SIZE_MAX, i));
                 ^
./src/basic/macro.h:182:19: note: in expansion of macro ‘__MIN’
 #define MIN(a, b) __MIN(UNIQ, (a), UNIQ, (b))
                   ^
src/resolve/test-resolved-packet.c:32:43: note: in expansion of macro ‘MIN’
                 assert_se(p->allocated >= MIN(DNS_PACKET_SIZE_MAX, i));
                                           ^
./src/basic/macro.h:187:60: warning: signed and unsigned type in conditional expression [-Wsign-compare]
                 UNIQ_T(A,aq) < UNIQ_T(B,bq) ? UNIQ_T(A,aq) : UNIQ_T(B,bq); \
                                                            ^
./src/basic/macro.h:44:44: note: in definition of macro ‘_unlikely_’
 #define _unlikely_(x) (__builtin_expect(!!(x),0))
                                            ^
./src/basic/macro.h:235:25: note: in expansion of macro ‘assert_message_se’
 #define assert_se(expr) assert_message_se(expr, #expr)
                         ^
src/resolve/test-resolved-packet.c:32:17: note: in expansion of macro ‘assert_se’
                 assert_se(p->allocated >= MIN(DNS_PACKET_SIZE_MAX, i));
                 ^
./src/basic/macro.h:182:19: note: in expansion of macro ‘__MIN’
 #define MIN(a, b) __MIN(UNIQ, (a), UNIQ, (b))
                   ^
src/resolve/test-resolved-packet.c:32:43: note: in expansion of macro ‘MIN’
                 assert_se(p->allocated >= MIN(DNS_PACKET_SIZE_MAX, i));
                                           ^
```
On the other hand, I understand what is being fixed here, so the warning can be fixed later. Thank you.